projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -416,9 +420,59 @@ internal void Write(ReadOnlyMemory<byte> memory)
 
         internal ValueTask WriteAsync(ReadOnlyMemory<byte> memory)
         {
+            SetNetworkTags(Activity.Current);
             return _frameHandler.WriteAsync(memory);
         }
 
+        private void SetNetworkTags(Activity? activity)
+        {
+            if (activity != null && activity.IsAllDataRequested)
+            {
+                switch (_frameHandler.RemoteEndPoint.AddressFamily)
+                {
+                    case AddressFamily.InterNetworkV6:
+                        activity.SetTag("network.type", "ipv6");
+                        break;
+                    case AddressFamily.InterNetwork:
+                        activity.SetTag("network.type", "ipv4");
+                        break;
+                }
+
+                if (!string.IsNullOrEmpty(_frameHandler.Endpoint.HostName))
+                {
+                    activity
+                        .SetTag("server.address", _frameHandler.Endpoint.HostName);
+                }
+
+                activity
+                    .SetTag("server.port", _frameHandler.Endpoint.Port);
+
+                if (_frameHandler.RemoteEndPoint is IPEndPoint ipEndpoint)
+                {
+                    string remoteAddress = ipEndpoint.Address.ToString();
+                    if (activity.GetTagItem("server.address") == null)
+                    {
+                        activity
+                            .SetTag("server.address", remoteAddress);
+                    }
+
+                    activity
+                        .SetTag("network.peer.address", remoteAddress)
+                        .SetTag("network.peer.port", ipEndpoint.Port);
+                }
+
+                if (_frameHandler.LocalEndPoint is IPEndPoint localEndpoint)
+                {
+                    string localAddress = localEndpoint.Address.ToString();
+                    activity
+                        .SetTag("client.address", localAddress)
+                        .SetTag("client.port", localEndpoint.Port)
+                        .SetTag("network.local.address", localAddress)
+                        .SetTag("network.local.port", localEndpoint.Port);
+                }
```

Comment:
it is possible for RabbitMQ to use something but AMQP  0.9.1? If so, it would be nice to populate `network.protocol.version`. I assume protocol name is always AMQP  and it does not make sense to ever populate it.